hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCompactionConfig.java

```diff
@@ -249,6 +249,21 @@
           + "record size estimate compute dynamically based on commit metadata. "
           + " This is critical in computing the insert parallelism and bin-packing inserts into small files.");
 
+  public static final ConfigProperty<String> ARCHIVE_MAX_FILES = ConfigProperty
+      .key("hoodie.archive.max.files")
+      .noDefaultValue()
+      .withDocumentation("The numbers of kept archive files under archived.");
+
+  public static final ConfigProperty<String> ARCHIVE_AUTO_TRIM_ENABLE = ConfigProperty
```

Comment:
Instead of deleting, can we coalesce them into bigger files and then delete the individual ones? Eventually, I am planning to make timeline just a single linear stream of instants, that is indexed. I am happy pen down the design as needed as well. But even in the interim, it might be good to not delete anything :)

Reply:
Sure, will compact these small archive files instead of deleting. This may need to do design work.
Will upgrade the code and design ASAP.
Thanks a lot for your review.